common/lib/client/auth.js

```diff
@@ -534,6 +519,144 @@ var Auth = (function() {
 		}
 	};
 
+	Auth.prototype._saveBasicOptions = function(defaultTokenParams, authOptions) {
+		this.method = 'basic';
+		this.key = authOptions.key;
+		this.basicKey = toBase64(authOptions.key);
+		this.tokenParams = defaultTokenParams || {};
+		this.authOptions = authOptions || {};
+		delete this.authOptions.force;
+		if('clientId' in authOptions) {
+			this._userSetClientId(authOptions.clientId);
+		}
+	}
+
+	Auth.prototype._saveTokenOptions = function(tokenParams, authOptions) {
+		this.method = 'token';
+
+		this.tokenParams = tokenParams || this.tokenParams || {};
+
+		/* If an authOptions object is passed in that contains new auth info (ie
+		* isn't just {force: true} or something), it becomes the new default, with
+		* the exception of the force attribute (RSA10g), which is set anew on each
+		* call to authorise (defaulting to false) */
+		this.force = false;
+		if(authOptions) {
+			this.force = authOptions.force;
+
+			if(containsAuthInfo(authOptions)) {
+				this.authOptions = authOptions || {};
```

Comment:
> if a user currently provides {} as an AuthOption arg in JS, it will reset AuthOption to empty

No it won't (with the implementation in this PR), see my previous comment.

Look, authOptions currently consists of only three things:

1. something which decides how to get a token (authUrl, authCallback, token, tokenDetails, or key). (what I called 'authInfo' above)
2. `force`
3. `queryTime`

For 1, there is only one operative way of getting a token at any one time, so IMHO merging just doesn't make sense - you never want an `authUrl` and an `authCallback` at the same time.

For 2, `force` explicitly doesn't persist either way, so merging vs replacing is irrelevant.

For 3, `queryTime` has no effect once the `serverTimeOffset` has been set once, so merging vs replacing is irrelevant.

So currently, the practical difference between "replace authOptions if passed-in once contains authInfo" and "only replace authInfo, merge all other options" is literally none at all.

Which means the only thing we're changing by making this policy decision now is the default for how any new authOptions we may add in the future will behave. Which I have no strong opinions on, I'll implement whatever you two decide. (Ditto on whether to replace/merge tokenParams).